week02/ValidParentheses.java

```diff
@@ -0,0 +1,30 @@
+// TC = O(n)
+// MC = O(n)
+// problem link : https://leetcode.com/problems/valid-parentheses/
+
+class Solution {
+
+    public boolean isValid(String s) {
+        Stack<Character> stk = new Stack<>();
+
+        for (int i = 0; i < s.length(); i++) {
+            char ch = s.charAt(i);
+
+            if (ch == '(' || ch == '{' || ch == '[') {
+                stk.push(ch);
+            } else if (stk.isEmpty()) {
+                return false; 
+            } else if (ch == ')' && stk.peek() != '(') {
+                return false;
+            } else if (ch == '}' && stk.peek() != '{') {
+                return false;
+            } else if (ch == ']' && stk.peek() != '[') {
+                return false;
+            } else {
+                stk.pop();
+            }
```

Comment:
Not a good way to solve this problem. What if I introduce a new kind of parentheses? You need to rewrite a lot of code again. Watch out the official solution as soon as I post them.

Reply:
Refactored as needed.